projects/ui-policy-studio-angular/src/lib/environment-flow-studio/gio-environment-flow-studio.component.spec.ts

```diff
@@ -37,82 +38,164 @@ describe('GioEnvironmentFlowStudioComponent', () => {
     await TestBed.configureTestingModule({
       imports: [GioEnvironmentFlowStudioComponent, NoopAnimationsModule, HttpClientTestingModule, MatIconTestingModule],
     }).compileComponents();
+  });
 
+  async function initComponent(apiType: ApiType, executionPhase: ExecutionPhase) {
     fixture = TestBed.createComponent(GioEnvironmentFlowStudioComponent);
     component = fixture.componentInstance;
-    environmentFlowStudioHarness = await TestbedHarnessEnvironment.harnessForFixture(fixture, GioEnvironmentFlowStudioHarness);
 
     component.policies = fakeAllPolicies();
     component.policySchemaFetcher = policy => of(fakePolicySchema(policy.id));
     component.policyDocumentationFetcher = policy => of(`${policy.id} documentation`);
+    component.apiType = apiType;
+    component.executionPhase = executionPhase;
     component.ngOnChanges({
       policies: new SimpleChange(null, null, true),
       policySchemaFetcher: new SimpleChange(null, null, true),
       policyDocumentationFetcher: new SimpleChange(null, null, true),
+      apiType: new SimpleChange(null, null, true),
+      executionPhase: new SimpleChange(null, null, true),
     });
-  });
+    environmentFlowStudioHarness = await TestbedHarnessEnvironment.harnessForFixture(fixture, GioEnvironmentFlowStudioHarness);
+  }
+
+  describe('PROXY API - REQUEST phase', () => {
+    beforeEach(async () => {
+      await initComponent('PROXY', 'REQUEST');
+    });
+
+    it('should display empty flow phase', async () => {
+      const flowPhase = await environmentFlowStudioHarness.getFlowPhase('REQUEST');
+
+      expect(flowPhase).toBeDefined();
+      const steps = await flowPhase.getSteps();
+
+      expect(steps).toEqual([
+        {
+          name: 'Incoming request',
+          type: 'connector',
+        },
+        {
+          name: 'Outgoing request',
+          type: 'connector',
+        },
+      ]);
```

Comment:
I think we miss some cases on response or subscribe/publish because on the story those info are not updated and I think they should